hw1.hs

```diff
@@ -0,0 +1,77 @@
+module Main where
+
+    data Lambda
+        = Var String         
+        | Fun String Lambda 
+        | App Lambda Lambda  -- application
+        deriving (Eq)
+
+    instance Show Lambda where
```

Comment:
This implementation shows certain terms in a weird way: 

```
ghci> App (Fun "x" (Var "x")) (Var "x")
(\x. x x)
```

The displayed string means an abstraction over `x` with the application of `x` to itself